algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+
+    ahorroTributario = (intereses + insuranceAmount + periodicCosts + depreciacion) * IR;
+    montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts) * IGV;
+    flujoBruto = cuota + insuranceAmount + periodicCosts;
+
+    if (inputs.buyingOption && periodo === periodos) {
+      montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts + montoRecompra) * IGV;
+      flujoBruto = cuota + insuranceAmount + periodicCosts + montoRecompra;
+      console.log(`Monto de recompra: S/ ${roundMoney(montoRecompra)}`);
+    }
+
+    flujoIGV = flujoBruto + montoIGVPeriodico;
+    flujoNeto = flujoBruto - ahorroTributario;
+
+    vnaFlujoBruto += flujoBruto / Math.pow(1 + descuentoKs, periodo);
+    vnaFlujoNeto += flujoNeto / Math.pow(1 + descuentoWACC, periodo);
+
+    console.log(`Saldo final: S/ ${roundMoney(saldoFinal)}`);
+    console.log(`Depreciación: S/ ${roundMoney(depreciacion)}`);
+    console.log(`Ahorro tributario: S/ ${roundMoney(ahorroTributario)}`);
+    console.log(`IGV: S/ ${roundMoney(montoIGVPeriodico)}`);
+    console.log(`Flujo Bruto: S/ ${roundMoney(flujoBruto)}`);
+    console.log(`Flujo con IGV: S/ ${roundMoney(flujoIGV)}`);
+    console.log(`Flujo Neto:S/ ${roundMoney(flujoNeto)}\n`);
+
+    totalAmortizacion = totalAmortizacion + amortizacion;
+
+    if (!(tipoPlazo === "Total")) {
+      totalIntereses = totalIntereses + intereses;
+    }
+
+    totalCostosPeriodicos += periodicCosts;
+    totalSeguro += insuranceAmount;
+    saldoInicial = saldoFinal;
+
+    flujosBrutos.push(roundMoney(flujoBruto));
+    flujosNetos.push(roundMoney(flujoNeto));
+  }
+
+  const desembolsoTotal = totalIntereses + totalAmortizacion + totalSeguro + totalCostosPeriodicos + montoRecompra;
+
+  console.log(`Monto total por intereses: S/ ${roundMoney(-totalIntereses)}`);
+  console.log(`Monto total amortizado: S/ ${roundMoney(-totalAmortizacion)}`);
+  console.log(`Monto total por costos periódicos: S/ ${roundMoney(-totalCostosPeriodicos)}`);
+  console.log(`Monto total por seguro contra todo riesgo: S/ ${roundMoney(-totalSeguro)}`);
+  console.log(`Desembolso total: S/ ${roundMoney(-desembolsoTotal)}\n`);
+
+  vnaFlujoBruto = vnaFlujoBruto + leasingAmount;
+  vnaFlujoNeto = vnaFlujoNeto + leasingAmount;
+  const tirFlujoBruto = irr(flujosBrutos, 0.01);
+  const tirFlujoNeto = irr(flujosNetos, 0.01);
+
+  const tceaFlujoBruto = Math.pow(1 + tirFlujoBruto, cuotasAnuales) - 1;
+  const tceaFlujoNeto = Math.pow(1 + tirFlujoNeto, cuotasAnuales) - 1;
+
+  console.log("= = = = = Indicadores de Rentabilidad = = = = =");
+  console.log(`TCEA Flujo Bruto: ${tceaFlujoBruto * 100}%`);
+  console.log(`TCEA Flujo Neto: ${tceaFlujoNeto * 100}%`);
+  console.log(`VAN Flujo Bruto: ${roundMoney(vnaFlujoBruto)}`);
+  console.log(`VAN Flujo Neto: ${roundMoney(vnaFlujoNeto)}`);
+  console.log(`TIR Flujo Bruto: ${tirFlujoBruto * 100}%`);
+  console.log(`TIR Flujo Neto: ${tirFlujoNeto * 100}%`);
```

Comment:
You can use the `Intl.NumberFormat` API here as well.

Reply:
Done